app/models/concerns/status_threading_concern.rb

```diff
@@ -51,12 +51,16 @@ def descendant_ids(limit, max_child_id, since_child_id, depth)
   end
 
   def descendant_statuses(limit, max_child_id, since_child_id, depth)
-    Status.find_by_sql([<<-SQL.squish, id: id, limit: limit, max_child_id: max_child_id, since_child_id: since_child_id, depth: depth])
+    # use limit + 1 and depth + 1 because 'self' is included
+    depth += 1 if depth.present?
+    limit += 1 if limit.present?
+
+    descendants_with_self = Status.find_by_sql([<<-SQL.squish, id: id, limit: limit, max_child_id: max_child_id, since_child_id: since_child_id, depth: depth])
       WITH RECURSIVE search_tree(id, path)
       AS (
         SELECT id, ARRAY[id]
         FROM statuses
-        WHERE in_reply_to_id = :id AND COALESCE(id < :max_child_id, TRUE) AND COALESCE(id > :since_child_id, TRUE)
+        WHERE id = :id AND COALESCE(id < :max_child_id, TRUE) AND COALESCE(id > :since_child_id, TRUE)
```

Comment:
Hm, I'm afraid that doesn't make much sense.
This will always select either the root status (if `max_child_id` and `since_child_id` or undefined or define a large enough range to include the root status), or no status at all, while it was supposed to select a subset of the direct replies to the current status.
So if my understanding is correct, `max_child_id` and `since_child_id` have been broken ever since this commit.

On the other hand, I'm not sure how useful `max_child_id` and `since_child_id` are for actually paginating… (especially since it's not currently used in a meaningful way anywhere in Mastodon)

Reply:
not sure I understand this comment. we're still recursing on in_reply_to_id, we're just starting with the the root status instead of replies to the root status)

Reply:
Yes it is supposed to be recursive, but since_child_id and max_child_id are supposed to apply to the direct children of the root toot, but it will be applied to the root toot instead.

Meaning this part of the query will either select the root status or no status at all. And the query itself will either return the whole thing, or no status at all, since_child_id and max_child_id are not used in a meaningful way.